data/database.py

```diff
@@ -2048,6 +2048,21 @@ class ManifestSecurityStatus(BaseModel):
     metadata_json = JSONField(default={})
 
 
+class ProxyCacheConfig(BaseModel):
+    """
+    Represents the configuration for an organization of type proxy cache
+    """
+
+    user_id = ForeignKeyField(User)
+    creation_date = DateTimeField(default=datetime.utcnow)
+    upstream_registry = CharField(max_length=2048)
+    upstream_registry_namespace = CharField(max_length=255, null=True)
```

Comment:
> Will we ask customers to provide it as two different inputs?
Not sure I follow. The namespace is optional, and should only be provided when caching a namespace upstream. When caching the whole registry the namespace should be omited. The UI needs to make this clear somehow.

> max_length seems to be chosen arbitrarily, upstream_registry is hostname + ':' + port, so 254+1+5=260.
True. We don't have any extra validation here - users might specify protocol (if they don't we default to https) or not. I'm okay with a smaller max_length though, 2048 does seem unnecessairily exceesive. What about 270, for a round nice number? It would also allow for specifying a protocol.

> Image references cannot have more than 255 characters: [...]
Hmm even if we do take the distribution implementation as the single source of truth, from what I gather it'd be hard to know exactly how big the namespace alone can be. A valid name could have a namespace 243 chars long and it would be valid. I can't really think of a better value here.

Reply:
I don't like the idea of splitting it into two different fields. We should ask customers to provide an image reference prefix. Like `docker.io`, or `docker.io/library`, or `quay.io/quay`. The same prefix they'll use in ImageContentSourcePolicy.
cc @dmesser

Reply:
Can you expand @dmage? I'm not seeing what the big deal is, or what's the relationship with ImageContentSourcePolicy.

Reply:
I mean, to have them separate makes sense to me because it makes it easier for us to distinguish in the code when orgs are set to cache whole registries vs single namespaces. We'd have to parse the URL to figure that out if the reference would be given as one, such as `docker.io/library`. Am I overcomplicating this?

Reply:
There's a bigger problem with using your suggestion @dmage. The `upstream_registry` is expected to be the api url of the registry, so in case of docker hub we want `registry-1.docker.io` rather than `docker.io`. We'd have to change that first in order to accept the namespace in that field.

Reply:
It makes it easier in the code, but I want to make it easier for people. I don't see any problems, other tools (docker, podman, CRI-O, skopeo, etc.) don't ask to provide `registry-1.docker.io`, they understand `docker.io`. As it's a simple splitting by / and translating one known string into another, we can do it in UI, if you want to go this way.

Reply:
I want to make it easier for people too. If you think your suggestion gives better usability then let's do it. I'll update the code.

Reply:
@dmage I believe I adressed your comments in this thread - marking the conversation as resolved. Lemme know if I missed something.